Utils/hdinsight-node-common/src/com/microsoft/azure/hdinsight/common/AbfsUri.java

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright (c) Microsoft Corporation
+ *
+ * All rights reserved.
+ *
+ * MIT License
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated
+ * documentation files (the "Software"), to deal in the Software without restriction, including without limitation
+ * the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and
+ * to permit persons to whom the Software is furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all copies or substantial portions of
+ * the Software.
+ *
+ * THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO
+ * THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+ * TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package com.microsoft.azure.hdinsight.common;
+
+import com.microsoft.azure.hdinsight.sdk.storage.StoragePathInfo;
+import org.apache.commons.lang3.StringUtils;
+
+import java.net.URI;
+import java.util.UnknownFormatConversionException;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class AbfsUri {
+    public static final Pattern ABFS_URI_PATTERN = Pattern.compile(
+            StoragePathInfo.AdlsGen2PathPattern, Pattern.CASE_INSENSITIVE);
+    public static final Pattern HTTP_URI_PATTERN = Pattern.compile(
+            StoragePathInfo.AdlsGen2RestfulPathPattern, Pattern.CASE_INSENSITIVE);
+
+    private final URI rawUri;
+    private final LaterInit<String> fileSystem = new LaterInit<>();
+    private final LaterInit<String> accountName = new LaterInit<>();
+    private final LaterInit<String> subPath = new LaterInit<>();
+
+    private AbfsUri(URI rawUri) {
+        this.rawUri = rawUri;
+    }
+
+    public URI getRawUri() {
+        return rawUri;
+    }
+
+    public String getFileSystem() {
+        return fileSystem.get();
+    }
+
+    public String getAccountName() {
+        return accountName.get();
+    }
+
+    public String getSubPath() {
```

Comment:
Let's align `getAbsolutePath` with `WasbUri`

Reply:
My first instinct is that absolute path means whole path or full path. But here what we need is relative path or sub path. Maybe we should replace the `getAbsolutePath` with `getSubPath`  in `WasbUri`.

---
In reply to: [361567180](https://github.com/microsoft/azure-tools-for-java/pull/3885#discussion_r361567180) [](ancestors = 361567180)